src/main/java/de/tum/cit/aet/artemis/programming/domain/ProgrammingExercise.java

```diff
@@ -815,6 +816,21 @@ public void validateSettingsForFeedbackRequest() {
         }
     }
 
+    /**
+     * Validates the network access feature for the given programming language.
+     * Currently, SWIFT and HASKELL do not support disabling the network access feature.
+     *
+     */
+    public void validateBuildPlanNetworkAccessForProgrammingLanguage() {
+        if (List.of(ProgrammingLanguage.SWIFT, ProgrammingLanguage.HASKELL).contains(getProgrammingLanguage())) {
```

Comment:
Do you expect the number of programming languages here to grow?
Otherwise I would prefer:
```suggestion
        if (ProgrammingLanguage.SWIFT == getProgrammingLanguage() || ProgrammingLanguage.HASKELL == getProgrammingLanguage()) {
```

Reply:
Yes, this may change in the future